lib/pact_broker/index/service.rb

```diff
@@ -194,6 +185,48 @@ def self.latest_verifications_for_consumer_version_tags(options)
           nil # should not be used
         end
       end
+
+      def self.query_pact_publication_ids_by_tags(base, tags)
+        latest = base.overall_latest
+        return latest.union(base.latest_for_consumer_tag(tags)) if tags.is_a?(Array)
+        return latest.union(base.latest_by_consumer_tag) if tags
+
+        latest
+      end
+
+      def self.base_query(options)
+        query = PactBroker::Pacts::PactPublication.select(Sequel[:pact_publications][:id])
+
+        if options[:consumer_name]
+          consumer = pacticipant_repository.find_by_name!(options[:consumer_name])
+          query = query.for_consumer(consumer)
+        end
+
+        if options[:provider_name]
+          provider = pacticipant_repository.find_by_name!(options[:provider_name])
+          query = query.for_provider(provider)
+        end
+
+        query
+      end
+
+
+      def self.pact_pacticipant_ids_by_name(pacticipant_name)
+        terms = pacticipant_name.split.map { |v| v.gsub("_", '\\_') }
+        string_match_query = Sequel.|(
+          Sequel.|( *terms.map { |term| Sequel.ilike(Sequel[:providers][:name], "%#{term}%") }),
+          Sequel.|( *terms.map { |term| Sequel.ilike(Sequel[:consumers][:name], "%#{term}%") })
+        )
+
+        PactBroker::Pacts::PactPublication
```

Comment:
I'm still confused as to why you're querying the pact publication table, instead of just querying the pacticipant table directly. To do this query requires that you join the pacticipant table twice to the pact publication table, and then filter on the entire dataset based on two columns (there could be 1000s of pact publications). 

If you query the pacticipant table directly, you only do the filtering once, there are no joins, and it's there are likely to be 10s of pacticipants maximum.

I feel like we're not on the same page here. Is there something I'm missing? Shall we have a lunch time catch up?